src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -116,8 +124,23 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
                 // Write blob headers
                 await blobClient.SetHttpHeadersAsync(CreateHttpHeaders(artifactSettings), cancellationToken: token);
 
-                // Write blob metadata
-                await blobClient.SetMetadataAsync(artifactSettings.Metadata, cancellationToken: token);
+                try
+                {
+                    Dictionary<string, string> mergedMetadata = artifactSettings.Metadata.Union(artifactSettings.CustomMetadata).ToDictionary(pair => pair.Key, pair => pair.Value);
```

Comment:
This will throw if two keys are the same. Consider either manually verifying that all of the keys are unique and then throw a specific exception if not or allowing custom metadata to either override the precalculated metadata or is dropped. For the latter cases, we should probably log warnings.

Reply:
Is the main reason for doing this so that we don't have to catch the `ArgumentException`? I had considered doing it this way but thought it might be slightly tedious, especially for something that probably will happen infrequently. That being said, I can do it this way, which would also allow us to keep custom metadata that _doesn't_ conflict (whereas right now, all custom metadata gets scrapped).

Reply:
In general, I try to do best-effort (unless it absolutely makes sense to abort an entire operation if a subset doesn't quite work).

We can easily determine if keys collide or not, and would make it much easier to understand from a user perspective what collision happened.